e2-studio-star-rx72n-firmware/src/tasks/bms_monitor_task.c

```
@@ -1163,32 +1340,21 @@ static void internal_bms_task_entry(ULONG input)
-      /* Check for low battery */
-      if (status.relative_soc < k_bms_low_soc_percent) {
-        rx_log_warn_val(s_tag, "Low battery SoC", (uint8_t)status.relative_soc);
-        (void)shared_data_set_event(k_event_low_battery);
-      }
-
-      /* Check for critical battery */
-      if (status.relative_soc < k_bms_critical_soc_percent) {
+      /* Check battery SoC thresholds - critical takes priority over warning.
+       * NOLINT(bugprone-branch-clone): branches differ in severity (e-stop vs. event),
+       * the structural similarity is intentional; suppressing false-positive. */
+      if (status.relative_soc < s_bms_config.soc_critical_pct) {
```

Clang (14.0.6)

[warning] repeated branch in conditional chain

(bugprone-branch-clone)